central/reprocessor/reprocessor.go

```diff
@@ -851,6 +854,47 @@ func (l *loopImpl) enrichImageV2(ctx context.Context, enrichCtx imageEnricher.En
 	return l.imageEnricherV2.EnrichImage(ctx, enrichCtx, image)
 }
 
+// getImageNamesWithSameSHA retrieves all known image names with the same SHA as the given image from the datastore.
+// This is used to ensure backward compatibility when sending image data to sensors that don't
+// have the FlattenImageDataOnSensor capability.
+func (l *loopImpl) getImageNamesWithSameSHA(img *storage.ImageV2) []*storage.ImageName {
+	if img.GetDigest() == "" {
+		return nil
+	}
+
+	// Query all images with this digest
+	query := search.NewQueryBuilder().AddExactMatches(search.ImageSHA, img.GetDigest()).
+		ForSearchResults(search.ImageRegistry, search.ImageRemote, search.ImageTag, search.ImageName).
+		ProtoQuery()
+	results, err := l.imagesV2.Search(allAccessCtx, query)
+	if err != nil {
+		log.Warnw("Failed to retrieve image names by digest",
+			logging.ImageName(img.GetName().GetFullName()),
+			logging.ImageID(img.GetId()),
+			logging.String("digest", img.GetDigest()),
+			logging.Err(err),
+		)
+		return nil
+	}
+
+	// Collect all image names
+	allNames := make([]*storage.ImageName, 0, len(results))
+	for _, res := range results {
+		if res.FieldValues != nil {
+			if nameVal, ok := res.FieldValues[strings.ToLower(search.ImageName.String())]; ok {
```

Comment:
Since we are doing this stuff in multiple places would it make sense to add a GetAllNames on the datastore and just use it?